### ISSUE TITLE
ComboBox: autofocus on first list item when ListBox is filtered (matching Autocomplete behaviour)

### ISSUE THREAD

**Issue by danielgavrilov** [adobe/react-spectrum#8007]
### Provide a general summary of the feature here

Autocomplete has a [`disableAutoFocusFirst` property](https://react-spectrum.adobe.com/react-aria/Autocomplete.html#props) which controls _"Whether or not to focus the first item in the collection after a filter is performed"_. 

It would be nice to have this for ComboBox as well.

### 🤔 Expected Behavior?

ComboBox supports a `disableAutoFocusFirst` property.

### 😯 Current Behavior

ComboBox doesn't support a `disableAutoFocusFirst` property.

### 💁 Possible Solution

Not sure if a patch would be possible or easy. I tried to adapt the code from `useAutocomplete` but didn't get it to work: https://github.com/adobe/react-spectrum/blob/3c9b36ae9e9bb42792ac4b3a51121969ae8d579c/packages/%40react-aria/autocomplete/src/useAutocomplete.ts#L167-L178

### 🔦 Context

I implemented a ComboBox for a country select, and I noticed users sometimes press Enter to select their country, which doesn't work because the first List item is not focussed by default when filtering. (The input field is also reset after pressing Enter, which makes the experience a bit worse.)

https://github.com/user-attachments/assets/c810f213-ff4e-41e3-b7cd-4cd2de1eade1

**Comment by abel-asfaw-rm** [comment]
+1 Did you ever find a solution for this @danielgavrilov

**Comment by LFDanLu** [comment]
Sorry for the late reply, https://github.com/adobe/react-spectrum/issues/5286#issuecomment-1773948662 has some relevant information as to why this wasn't the behavior for ComboBox in the first place. It is worth a reinvestigation though since screen reader behaviors can change over time.
